administrator/components/com_media/models/api.php

```diff
@@ -38,85 +46,126 @@ public function __construct($config = array())
 	{
 		parent::__construct($config);
 
+		if (!isset($config['providers']))
+		{
+			$config['providers'] = Joomla\CMS\Plugin\PluginHelper::getPlugin('filesystem');
+			$this->providers = $config['providers'];
+		}
+
 		if (!isset($config['fileadapter']))
 		{
-			// Import Local file system plugin
-			JPluginHelper::importPlugin('filesystem');
+			// Import enabled file system plugins
+			Joomla\CMS\Plugin\PluginHelper::importPlugin('filesystem');
 
 			$app = JFactory::getApplication();
 
 			$results = $app->triggerEvent('onFileSystemGetAdapters');
+			$adapters = array();
 
-			if ($results != null)
+			for ($i = 0, $len = count($results); $i < $len; $i++)
 			{
-				$config['fileadapter'] = $results[0];
+				$adapters[$this->providers[$i]->name] = $results[$i];
 			}
+
+			$config['fileadapter'] = $adapters;
+			$this->adapters = $adapters;
```

Comment:
Same as above: What happens when config['fileadapter'] is set?
This should be moved outside of the if block to allow setting fileadapter via config.

Additionaly the naming should be improved: you use singular config['fileadapter'] and set it to plural $adapters.